### PR TITLE
Bump to 0.76.0; update changelog

### PR DESCRIPTION
**Pull request by weihanglo**
[rendered](https://github.com/weihanglo/cargo/blob/version-bump/CHANGELOG.md)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -19,6 +153,83 @@
...
+  [#12733](https://github.com/rust-lang/cargo/pull/12733)
+- mdbook: use *AND* search when having multiple terms.
+  [#12548](https://github.com/rust-lang/cargo/pull/12548)
+- Established publish best practices 
+  [#12745](https://github.com/rust-lang/cargo/pull/12745)
+- Clarify caret requirements.
+  [#12679](https://github.com/rust-lang/cargo/pull/12679)
+- Clarify how `version` works for `git` dependencies.
+  [#12270](https://github.com/rust-lang/cargo/pull/12270)
+- Carify and differentiate defaults for split-debuginfo.
```
Comment (ehuss) [inline CHANGELOG.md:171]:
```suggestion
- Clarify and differentiate defaults for split-debuginfo.
```

**Comment by weihanglo** [comment]
I guess I need a Grammarly-like tool…

Just fixed them all. Thanks for the review!

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,16 +1,122 @@
...
+  [#12591](https://github.com/rust-lang/cargo/pull/12591)
+  [#12614](https://github.com/rust-lang/cargo/pull/12614)
+
 ### Changed
 
+- ❗️ Changed how arrays in configuration are merged.
+  The order was unspecified and now follows how other configuration types work for consistency.
+  [summary](https://blog.rust-lang.org/inside-rust/2023/08/24/cargo-config-merging.html)
+  [#12515](https://github.com/rust-lang/cargo/pull/12515)
+- cargo-update: renamed option `--aggressive` to `--recursive`.
```
Comment (epage) [inline CHANGELOG.md:80]: nit: people might read into this that its a breaking change and get worried.  Maybe instead "silently deprecate `--aggressive` in favor of the new `--recursive`

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,16 +1,122 @@
...
+  [#12544](https://github.com/rust-lang/cargo/pull/12544)
+- cargo-install: suggest `--git` when the package name looks like a URL.
+  [#12575](https://github.com/rust-lang/cargo/pull/12575)
+- cargo-add: display either a feature list or a summarized count.
+  [#12662](https://github.com/rust-lang/cargo/pull/12662)
+  [#12702](https://github.com/rust-lang/cargo/pull/12702)
+- cargo-clean: error out if `--doc` is mixed with `-p`.
+  [#12637](https://github.com/rust-lang/cargo/pull/12637)
+- `-p/--package` can be used as a positional argument in most of the cargo commands.
```
Comment (epage) [inline CHANGELOG.md:89]: This change only affects `update`.  Most cargo commands still require `-p`.  This makes `update` consistent with `add` and `remove`.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,16 +1,122 @@
...
+  [#12556](https://github.com/rust-lang/cargo/pull/12556)
+- Enhanced filesystem error by adding wrappers around `std::fs::metadata`.
+  [#12636](https://github.com/rust-lang/cargo/pull/12636)
+- Enhanced resolver version mismatch warning.
+  [#12573](https://github.com/rust-lang/cargo/pull/12573)
+- Use clap to suggest alternative argument for unsupported arguments.
+  [#12529](https://github.com/rust-lang/cargo/pull/12529)
+  [#12693](https://github.com/rust-lang/cargo/pull/12693)
+  [#12723](https://github.com/rust-lang/cargo/pull/12723)
+- Removed redundant information from cargo new/init.
```
Comment (epage) [inline CHANGELOG.md:117]: nit: this should probably have the context that this is for the `--help` output

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,16 +1,122 @@
...
+  [summary](https://blog.rust-lang.org/inside-rust/2023/08/24/cargo-config-merging.html)
+  [#12515](https://github.com/rust-lang/cargo/pull/12515)
+- cargo-update: renamed option `--aggressive` to `--recursive`.
+  [#12544](https://github.com/rust-lang/cargo/pull/12544)
+- cargo-install: suggest `--git` when the package name looks like a URL.
+  [#12575](https://github.com/rust-lang/cargo/pull/12575)
+- cargo-add: display either a feature list or a summarized count.
+  [#12662](https://github.com/rust-lang/cargo/pull/12662)
+  [#12702](https://github.com/rust-lang/cargo/pull/12702)
+- cargo-clean: error out if `--doc` is mixed with `-p`.
```
Comment (epage) [inline CHANGELOG.md:87]: Should this have a "❗️"?

**Comment by weihanglo** [comment]
Thanks! Every review comment is addressed now I believe.

**Pull request merged**